### DISCUSSION TITLE
Having issues rendering templates using jinja

### DISCUSSION THREAD

**Discussion by lurdfab** [pallets/flask#4974] [Q&A]
Please I don't know if anyone can be of assistance to me cos I am having a challenge and I have done everything possible to solve it but still no improvements, I'm actually trying to render the templates using jinja2 the index page which is the home page is working well but other pages aren't coming up,Like the signup, login and contact us page, the header and footer renders very well but the content which is the form doesn't show or render, I have tried to do it several times for days now and i'm still getting the same results, I would be so glad if anyone can help out though, thanks.

**Comment by ThiefMaster** [comment]
unless your provide more information beyond just a vague description of your problem, nobody will be able to help you...

post some code (or link to your git repo) and add a screenshot for a starter...

**Comment by lurdfab** [comment]
![Screenshot from 2023-02-11 09-46-27](https://user-images.githubusercontent.com/119491907/218249287-c5ab1828-f10e-4069-b993-31a8476508b3.png)
![Screenshot from 2023-02-11 09-47-38](https://user-images.githubusercontent.com/119491907/218249291-035542d4-e910-43d2-b1e2-1f953459e414.png)
![Screenshot from 2023-02-11 09-49-04](https://user-images.githubusercontent.com/119491907/218249293-66ac36b0-8e90-4103-84ea-8b3a54edc80d.png)

**Comment by lurdfab** [comment]
These are the screenshots of the codes, please I need assistance

**Comment by lurdfab** [comment]
[image: Screenshot from 2023-02-11 09-46-27.png][image: Screenshot from
2023-02-11 09-47-38.png][image: Screenshot from 2023-02-11 09-49-04.png]

These are screenshots of the codes, I have also sent it to the discussion
group

**Reply by ThiefMaster** [reply]
post code as text. i meant screenshots showing what's going wrong

**Comment by lurdfab** [comment]
```python
from blog import app
from flask import render_template

@app.route("/")
def index():
    return render_template ("index.html")

@app.route("/contact-us")
def contact_us():
    return render_template ("contact_us.html")

@app.route("/signup")
def signup():
    return render_template ("signup.html")
```

That is for the routes

```html
{% extends 'base.html' %}
 

{% block header %}
<div class="container">
    <header class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">
        <a href="{{ url_for ('index') }}" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
            <img src="{{ url_for ('static', filename='img/logo.jpg') }}" height="70px" alt="">
            <h5>Legit News</h5>
        </a>
    </header>
</div>
{% endblock %}
<!-- header done  -->

<!-- contact us  -->
<div class="container">
    <form class="form-signup">
      
      <h1 class="h3 mb-3 fw-normal">Fill in the Fields</h1>
      <div class="row">
        <div class="col-md-6">
          <div class="form-floating">
            <input type="text" class="form-control" id="floatingInput" placeholder="full names here">
            <label for="floatingInput">First Name</label>
          </div>
          <br>
        </div>
        <div class="col-md-6">
          <div class="form-floating">
            <input type="text" class="form-control" id="floatingInput" placeholder="full names here">
            <label for="floatingInput">Last Name</label>
          </div>
          <br>
        </div>
        <div class="form-group">
          <div class="form-floating">
            <input type="email" class="form-control" id="floatingInput" placeholder="name@example.com">
            <label for="floatingInput">Email address</label>
          </div>
          <br>
        </div>
        <div class="form-group">
          <div class="form-floating textarea">
            <textarea class="form-control" placeholder="Leave a comment here" id="floatingTextarea2" style="height: 100px"></textarea>
            <label for="floatingTextarea2">Comments</label>
          </div>
          <br>
        </div>
        <div class="but">
          <input type="submit" class="btn btn-secondary btn-lg value="submit">
        </div>
      </div>    
    </form>
</div>
<!-- contact us done  -->

<!-- footer  -->
{% block footer %}{% include 'footer.html' %}{% endblock %}
<!-- footer done  --> 
```

this is for one of the pages that isn't popping up above

![Screenshot from 2023-02-11 09-57-48](https://user-images.githubusercontent.com/119491907/218249660-f8520ce6-06ef-4c7e-8125-f09cf75ed8c8.png)


The result when it loads, renders the header and footer but not the content which is the form

**Reply by ThiefMaster** [reply]
(come on, you're a developer and didn't realize somethign is wrong when you pasted it here and it all looked unformatted? i updated your post so it's properly monospaced and highlighted...)

**Reply by lurdfab** [reply]
Thanks mate, I'm a newbie in this field bare a month as a python developer
